src/PKSim.Core/Snapshots/Mappers/SnapshotMapper.cs

```diff
@@ -7,6 +7,37 @@
 
 namespace PKSim.Core.Snapshots.Mappers
 {
+   public class SnapshotContext
+   {
+      public int Version { get; }
+      public PKSimProject Project { get; }
+
+      //This constructor should only be called when initiation the project load and the project is not available
+      public SnapshotContext() : this(new PKSimProject(), ProjectVersions.Current)
+      {
+      }
+
+      public SnapshotContext(PKSimProject project, int version)
+      {
+         Project = project;
+         Version = version;
+      }
+
+      public SnapshotContext(SnapshotContext baseContext) : this(baseContext.Project, baseContext.Version)
+      {
+      }
+
+      /// <summary>
+      /// Returns true if the format is V9 or earlier
+      /// </summary>
+      public bool IsV9FormatOrEarlier => Version <= ProjectVersions.V9;
```

Comment:
This is how we know what conversion needs to happen. Much better than checking fro properties etc..added or removed